File: Scala/InsertionSort.scala
import scala.collection.mutable.ListBuffer

object Main extends App{

    println(insertionSort(ListBuffer(5, 2, 4, 6, 1, 3)));
    
    /**
     * 挿入ソートを実装
     * @param list リスト
     **/
    def insertionSort(list: ListBuffer[Int]):ListBuffer[Int] = {
        for (i <- 0 until list.length) {
            var beSortedValue = list.apply(i);
            var j = i - 1;
            while(j >= 0 && list.apply(j) > beSortedValue) {
                list.update(j + 1, list.apply(j));
                list.update(j, beSortedValue);
                j -= 1;
            }
        }
        return list
    }
}

File: Dart/InsertionSort.dart
void main() {
  
  /**
   * 挿入ソート
   * @param list リスト
   */
  List<int> insertionSort(List<int> list) {
    for (var i = 0; i < list.length; i++) {
      var beSortedValue = list[i];
      var j = i - 1;
      while(j >= 0 && list[j] > beSortedValue) {
        list[j + 1] = list[j];
        list[j] = beSortedValue;
        j--;
      }
    }
    return list;
  }
  
  print(insertionSort([5, 2, 4, 6, 1, 3]));
}
